iree/compiler/Conversion/LinalgToLLVM/ConvertToLLVM.cpp

```diff
@@ -0,0 +1,159 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iree/compiler/Conversion/LinalgToLLVM/Passes.h"
+#include "iree/compiler/Dialect/HAL/IR/HALDialect.h"
+#include "iree/compiler/Dialect/HAL/IR/HALOps.h"
+#include "iree/compiler/Dialect/IREE/IR/IREEOps.h"
+#include "iree/compiler/Dialect/Shape/IR/ShapeOps.h"
+#include "mlir/Conversion/AffineToStandard/AffineToStandard.h"
+#include "mlir/Conversion/LinalgToLLVM/LinalgToLLVM.h"
+#include "mlir/Conversion/SCFToStandard/SCFToStandard.h"
+#include "mlir/Conversion/StandardToLLVM/ConvertStandardToLLVM.h"
+#include "mlir/Conversion/StandardToLLVM/ConvertStandardToLLVMPass.h"
+#include "mlir/Conversion/VectorToLLVM/ConvertVectorToLLVM.h"
+#include "mlir/Conversion/VectorToSCF/VectorToSCF.h"
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+#include "mlir/Dialect/StandardOps/IR/Ops.h"
+#include "mlir/Pass/Pass.h"
+
+namespace mlir {
+namespace iree_compiler {
+
+// Converts tie_shape op to an insert shape and stride information into Memref
+// descritor.
+class ConvertTieShapePattern : public ConvertToLLVMPattern {
+ public:
+  explicit ConvertTieShapePattern(MLIRContext *context,
+                                  LLVMTypeConverter &lowering_)
+      : ConvertToLLVMPattern(Shape::TieShapeOp::getOperationName(), context,
+                             lowering_) {}
+
+  LogicalResult matchAndRewrite(
+      Operation *op, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const override {
+    auto tieShapeOp = cast<Shape::TieShapeOp>(op);
+    auto loc = tieShapeOp.getLoc();
+    MemRefDescriptor sourceMemRef(operands.front());
+    auto makeRankedShapeOp =
+        cast<Shape::MakeRankedShapeOp>(tieShapeOp.shape().getDefiningOp());
+    auto rankedShapeType = makeRankedShapeOp.shape()
+                               .getType()
+                               .dyn_cast_or_null<Shape::RankedShapeType>();
+    if (!rankedShapeType) return failure();
+
+    auto shape = rankedShapeType.getAllDims();
+
+    // Update memref descriptor shape and strides.
+    for (int i = 0; i < shape.size(); ++i) {
+      if (shape[i] == -1) {
+        sourceMemRef.setSize(rewriter, loc, i,
+                             makeRankedShapeOp.dynamic_dimensions()[i]);
+      } else {
+        sourceMemRef.setConstantSize(rewriter, loc, i, shape[i]);
+      }
+    }
+    // Compute and update memref descriptor strides,
```

Comment:
Is it always correct to use natural size and stride? This need to be consistent with the runtime side. Would be nice to at least have a comment here to make sure we remind ourselves.